### PR TITLE
Discovery rules permissions tests from UI and org/location association tests via api

### PR DESCRIPTION
**Pull request by sghai**
Test Results:
**Org Tests**

```console
]# py.test -v tests/foreman/api/test_discoveryrule.py::DiscoveryRuleTestCase -k test_positive_create_with_org_loc================================================================ test session starts =================================================================
platform linux2 -- Python 2.7.12, pytest-3.0.2, py-1.4.31, pluggy-0.3.1 -- /usr/bin/python
cachedir: .cache
rootdir: /home/sghai/QE/robottelo, inifile: 
collected 12 items 

tests/foreman/api/test_discoveryrule.py::DiscoveryRuleTestCase::test_positive_create_with_org_loc <- robottelo/decorators/__init__.py PASSED

================================================================ 11 tests deselected =================================================================
====================================================== 1 passed, 11 deselected in 20.51 seconds ======================================================
```
```console
[ robottelo]# py.test -v tests/foreman/api/test_discoveryrule.py::DiscoveryRuleTestCase -k test_positive_update_org_loc================================================================ test session starts =================================================================
platform linux2 -- Python 2.7.12, pytest-3.0.2, py-1.4.31, pluggy-0.3.1 -- /usr/bin/python
cachedir: .cache
rootdir: /home/sghai/QE/robottelo, inifile: 
collected 12 items 

tests/foreman/api/test_discoveryrule.py::DiscoveryRuleTestCase::test_positive_update_org_loc <- robottelo/decorators/__init__.py PASSED

================================================================ 11 tests deselected =================================================================
====================================================== 1 passed, 11 deselected in 23.75 seconds ======================================================
```

**Roles Tests**
```console
robottelo]$ py.test -v tests/foreman/ui/test_discoveryrule.py::DiscoveryRuleReaderRoleTestCase
================================================================ test session starts =================================================================
platform linux2 -- Python 2.7.12, pytest-3.0.2, py-1.4.31, pluggy-0.3.1 -- /usr/bin/python
cachedir: .cache
rootdir: /home/sghai/QE/robottelo, inifile: 
collected 3 items 

tests/foreman/ui/test_discoveryrule.py::DiscoveryRuleReaderRoleTestCase::test_negative_delete_rule_with_non_admin_user <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_discoveryrule.py::DiscoveryRuleReaderRoleTestCase::test_positive_list_host_based_on_rule_search_query <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_discoveryrule.py::DiscoveryRuleReaderRoleTestCase::test_positive_view_existing_rule_with_non_admin_user <- robottelo/decorators/__init__.py PASSED

======================================================== 2 passed, 1 skipped in 68.74 seconds ========================================================
```
```console
robottelo]$ py.test -v tests/foreman/ui/test_discoveryrule.py::DiscoveryRuleManagerRoleTestCase
================================================================ test session starts =================================================================
platform linux2 -- Python 2.7.12, pytest-3.0.2, py-1.4.31, pluggy-0.3.1 -- /usr/bin/python
cachedir: .cache
rootdir: /home/sghai/QE/robottelo, inifile: 
collected 2 items 

tests/foreman/ui/test_discoveryrule.py::DiscoveryRuleManagerRoleTestCase::test_positive_create_rule_with_non_admin_user <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_discoveryrule.py::DiscoveryRuleManagerRoleTestCase::test_positive_delete_rule_with_non_admin_user <- robottelo/decorators/__init__.py PASSED

============================================================= 2 passed in 76.72 seconds ==============================================================
```

### CODE REVIEW

File: tests/foreman/ui/test_discoveryrule.py [hunk 1]
```diff
@@ -711,13 +816,18 @@ def test_positive_view_existing_rule_with_non_admin_user(self):
...
 
-        @caseautomation: notautomated
-
         @CaseLevel: Integration
         """
+        with Session(
+            self.browser, self.username, self.user_password
+        ) as session:
+            session.nav.go_to_discovery_rules()
+            locators['discoveryrules.disabled_name']
```
Comment (rochacbruno) [inline tests/foreman/ui/test_discoveryrule.py]: Is this line required? it is just accessing locator but not using this.

Comment (sghai) [inline tests/foreman/ui/test_discoveryrule.py]: Nop.. its not required at all.. somehow I missed to remove.. thank you for pointing that.

File: tests/foreman/ui/test_discoveryrule.py [hunk 2]
```diff
@@ -689,13 +737,70 @@ def test_positive_delete_rule_with_non_admin_user(self):
...
+        # Create a discovery rule
+        cls.rule_name = gen_string('alpha')
+        entities.DiscoveryRule(
+            name=cls.rule_name,
+            hostgroup=cls.host_group,
+            search_='cpu_count = 1',
+            organization=[cls.org],
+            location=[cls.loc],
+        ).create()
+        # create non-admin user with Discovery_Manager role
```
Comment (oshtaier) [inline tests/foreman/ui/test_discoveryrule.py]: `Discovery Reader`

File: tests/foreman/ui/test_discoveryrule.py [hunk 3]
```diff
@@ -727,10 +836,17 @@ def test_negative_delete_rule_with_non_admin_user(self):
...
         @CaseLevel: Integration
         """
+        with Session(
+            self.browser, self.username, self.user_password
+        ) as session:
+            session.nav.go_to_discovery_rules()
+            element = self.discoveryrules.find_element(
+                locators['discoveryrules.disabled_name'] % self.rule_name)
+            self.assertIsNotNone(element)
+            with self.assertRaises(UIError):
```
Comment (oshtaier) [inline tests/foreman/ui/test_discoveryrule.py]: `UIError` is a pretty global class, can we check for some exact error message too

**Comment by oshtaier** [comment]
To be honest, I think that these setups for different classes are little bit redundant and probably we can unite them into something one. It is just so small difference between them. I understand logical aspect, but anyway, could you please think about that?

**Comment by sghai** [comment]
@oshtaier : Sure.. I'll try to combine them..

**Comment by sghai** [comment]
Test Results:

```console
 robottelo]$ py.test -v tests/foreman/ui/test_discoveryrule.py::DiscoveryRuleRoleTestCase
============================= test session starts ==============================
platform linux2 -- Python 2.7.12, pytest-3.0.2, py-1.4.31, pluggy-0.3.1 -- /usr/bin/python
cachedir: .cache
rootdir: /home/sghai/QE/robottelo, inifile: 
collected 5 items 

tests/foreman/ui/test_discoveryrule.py::DiscoveryRuleRoleTestCase::test_negative_delete_rule_with_non_admin_user <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_discoveryrule.py::DiscoveryRuleRoleTestCase::test_positive_create_rule_with_non_admin_user <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_discoveryrule.py::DiscoveryRuleRoleTestCase::test_positive_delete_rule_with_non_admin_user <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_discoveryrule.py::DiscoveryRuleRoleTestCase::test_positive_list_host_based_on_rule_search_query <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_discoveryrule.py::DiscoveryRuleRoleTestCase::test_positive_view_existing_rule_with_non_admin_user <- robottelo/decorators/__init__.py PASSED

==================== 4 passed, 1 skipped in 113.74 seconds =====================
```

**Comment by oshtaier** [comment]
ACK

**Pull request merged**